test/proposal_settings_test.rb

```diff
@@ -0,0 +1,196 @@
+    }
+  end
+  let(:use_vnc) { false }
+  let(:use_ssh) { false }
+
+  before do
+    allow(Yast::Linuxrc).to receive(:vnc).and_return(use_vnc)
+    allow(Yast::Linuxrc).to receive(:usessh).and_return(use_ssh)
+
+    allow(Yast::ProductFeatures).to receive("GetSection")
+      .with("globals").and_return(global_section)
+  end
+
+  describe "#initialize" do
+    it "loads the default values from the control file" do
+      expect_any_instance_of(described_class).to receive(:load_features)
+
+      described_class.create_instance
+    end
+
+    context "when firewall has been enabled in the control file" do
+      let(:global_section) { { "enable_firewall" => true, "enable_sshd" => false } }
+
+      it "sets firewalld service to be enabled" do
+        expect_any_instance_of(described_class).to receive(:enable_firewall!)
+
+        described_class.create_instance
+      end
+    end
+
+    context "when ssh has been enable by Linuxrc" do
+      let(:use_ssh) { true }
+
+      it "sets ssh service to be enabled" do
+        expect_any_instance_of(described_class).to receive(:enable_sshd!)
+
+        described_class.create_instance
+      end
+
+      it "sets the ssh port to be opened" do
+        expect_any_instance_of(described_class).to receive(:open_ssh!)
+
+        described_class.create_instance
+      end
+    end
+
+    context "when vnc has been enable by Linuxrc" do
+      let(:use_vnc) { true }
+
+      it "sets the vnc port to be opened" do
+        expect_any_instance_of(described_class).to receive(:open_vnc!)
+
+        described_class.create_instance
+      end
+    end
+  end
+
+  describe "#enable_firewall!" do
+    it "sets firewalld service to be enabled" do
+      allow(Yast::PackagesProposal).to receive("AddResolvables")
+        .with("firewall", :package, ["firewalld"])
+
+      expect(subject.enable_firewall).to be(false)
+      subject.enable_firewall!
+      expect(subject.enable_firewall).to be(true)
+    end
+
+    it "adds the firewalld package to be installed" do
+      expect(Yast::PackagesProposal).to receive("AddResolvables")
+        .with("firewall", :package, ["firewalld"])
+
+      subject.enable_firewall!
+    end
+  end
+
+  describe "#disable_firewall!" do
+    it "sets firewalld service to be disabled" do
+      allow(Yast::PackagesProposal).to receive("RemoveResolvables")
+        .with("firewall", :package, ["firewalld"])
+
+      subject.disable_firewall!
+      expect(subject.enable_firewall).to be(false)
+    end
+
+    it "removes the firewalld package for current selection" do
+      expect(Yast::PackagesProposal).to receive("RemoveResolvables")
+        .with("firewall", :package, ["firewalld"])
+
+      subject.disable_firewall!
+    end
+  end
+
+  describe "#enable_sshd!" do
+    it "sets sshd service to be enabled" do
+      allow(Yast::PackagesProposal).to receive("AddResolvables")
+        .with("firewall", :package, ["openssh"])
+
+      subject.enable_sshd!
```

Comment:
when seeing this, what is reason to have these methods with "!" ?

Reply:
well if it changes the internal state of the object then the convention is to use "!" isn't it?

Reply:
well, see discussion at https://stackoverflow.com/questions/7179016/what-is-the-purpose-of-and-at-the-end-of-method-names
If it is surprising or dangerous operation it should have "!". For me it indicates like it it try to enable it now and can raise if failed ( but maybe just my POV, we can ask someone else how he see it ). @mvidner @lslezak or @imobachgs ?

Reply:
I'm fine with the discussion. As this implementation was mainly for the POC, what if we maintain the discussion and maybe change the design but merge this PR that is now covering with unit tests the code and then apply the discussion results in next PR.

Are you fine with that?

Reply:
And well about surprising or dangerous, it is not always the case: usually the exclamation mark (!)  means the object is modified like with `downcase!` `sort!` and so on while `downcase` `sort`don't modify the object.